pennylane/pulse/rydberg_hamiltonian.py

```diff
@@ -218,29 +218,42 @@ def circuit_local(params):
             qml.evolve(H)(params, t=[0, 10])
             return qml.expval(qml.PauliZ(0))
 
-    >>> params = [2.4, [1.3, -2.0]]
+    >>> params = [2.4, [1.3, -2.0], -1.5]
     >>> circuit_local(params)
-    Array(0.45782223, dtype=float64)
+    Array(0.80238028, dtype=float64)
     >>> jax.grad(circuit_local)(params)
-    [Array(-0.33522988, dtype=float64),
-     [Array(0.40320718, dtype=float64, weak_type=True),
-      Array(-0.12003976, dtype=float64, weak_type=True)]]
+    [Array(-0.73273312, dtype=float64),
+     [Array(-0.45635171, dtype=float64, weak_type=True),
+      Array(0.76572817, dtype=float64, weak_type=True)],
+     Array(0.01027312, dtype=float64)]
     """
     if isinstance(wires, int):
         wires = [wires]
+    trivial_detuning = not callable(detuning) and qml.math.isclose(detuning, 0.0)
 
-    # We compute the `coeffs` and `observables` of the laser field
-    coeffs = [
-        amplitude_and_phase(qml.math.cos, amplitude, phase),
-        amplitude_and_phase(qml.math.sin, amplitude, phase),
-        detuning,
-    ]
+    if not callable(amplitude) and qml.math.isclose(amplitude, 0.0):
+        if trivial_detuning:
+            raise ValueError("The global drive must have a non-trivial amplitude.")
```

Comment:
```suggestion
            raise ValueError(f"Expected non-zero value for at least one of either amplitude or detuning, but received amplitude={amplitude} and detuning={detuning}. All terms are zero.")
```
It's confusing that this seems to imply I am applying a global drive if I set detuning equal to zero. The limitation that local Rydberg drive is only detuning is only for the Aquila hardware, not a philosophical limitation on Rydberg systems.